pkmodel/solution.py

```diff
@@ -7,11 +7,76 @@ class Solution:
 
     Parameters
     ----------
-
-    value: numeric, optional
-        an example paramter
+    
+    model: an object in the Model Class
+    protocol: an object in the Protocol Class
+    T: The system is solved for the time interval [0, T]
 
     """
-    def __init__(self, value=44):
-        self.value = value
+    def __init__(self, model, protocol, T):
+        self.model, self.protocol, self.T = model, protocol, T
 
+    # Solve the system of ODEs
+    def solve(self):
+    
+        # Build up the system of ODEs
+        if self.model.dosing_type == "intravenous":
+        
+            # Assuming q = [q1, ..., qn, qc], Q = [Q1, ..., Qn], V = [V1, ..., Vn, Vc]
+            def f(t, q, Q, V):
+                dydt = []
+                
+                # Differential equations for the peripheral compartments q1, ..., qn
+                for i in [0:self.model.number_of_compartments]:
+                    dydt.append(Q[i]*(q[-1]/V[-1] - q[i]/V[i])
+                
+                # Differential equation for the central compartment qc
+                # Assuming protocol is the Dose(t) function
+                # ??? Sum can be used as below?
+                dydt.append(self.protocol - (q[-1]/V[-1]) * self.model.CL - sum(dydt))
+                return dydt
+        
+            t_span = (0, T)
```

Comment:
Comment what is t_span?
            t_span = (0, T)

Reply:
And in general a bit more clear, probably say more in the docstring not to avoid getting confused